horace_core/sqw/PixelData/@PixelData/get_pix_in_ranges.m

```diff
@@ -13,6 +13,9 @@
 % -------
 % pix_out     A PixelData object containing the pixels in the given ranges.
 %
+if ~exist('recalculate_pix_range','var')
```

Comment:
checking this again - online matlab docs say if the last arg is missing the call will be flagged as incomplete args. What determines that this arg "does not exist"?

Reply:
it looks like Matlab allocates and assigns the actual variables values according to the list of the input arguments. If some arguments are missing, the variables, corresponding to the arguments are undefined. The equivalent to this would be `varargin` and `nargin` check, but I have not seen difference in speed and direct mention of the missing variable (if only one may be missing) reads better.